### ISSUE TITLE
New Layout/AlignHash has a runtime error on mixed Hash style

### ISSUE THREAD

**Issue by kmoschcau** [rubocop-hq/rubocop#7066]
## Expected behavior

Rubocop should run the Layout/AlignHash cop without an error occuring.

## Actual behavior

There occurs an error with the message "nil can't be coerced into Integer"

## Steps to reproduce the problem

Minimal example:
```ruby
bar(
  foo:  'baz',
  'bar' => 'bla'
)
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.70.0 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-linux)
```

**Comment by rmm5t** [comment]
I just ran into this as well, here was our specific use-case:

```ruby
      {
        'Content-Type' => 'application/json',
        Authorization: "Bearer #{@access_token}",
      }
```

**Comment by rmm5t** [comment]
PR ready for review here: #7068
